### ISSUE TITLE
About code implementation

### ISSUE THREAD

**Issue by liushengyan** [youngzhou97qz/CGMVQA#1]
hi, I met some problem that I didn't get 0.64 accuracy after running this code, I alse set the parameters according to the paper right, like epoch = 40, batch_size = 64, but I got 0.6 accuracy only at most, I thought maybe you knew what the problem was, thank you very much.

**Comment by VirajBagal** [comment]
Hi @liushengyan , I am having same experience as you. Using their code (where they have kept the resnet parameters non-trainable), I managed to get only 0.5658. What did you do to get 0.6 accuracy? (Resnet weights trainable? What augmentations?) @youngzhou97qz Can you please comment on what extra needs to be done to get 0.64?

**Comment by youngzhou97qz** [comment]
Thanks for your comment. I'm not sure what is different in your implementation. Different optimizers, learning rates, the order of layers, dim sizes, weights sharing or not, dropout rates, ensemble methods, random number seeds, etc. will affect the results. Each type of model is trained separately, and the required epochs are not the same.

**Comment by liushengyan** [comment]
Oh, thanks for your reply.&nbsp;Could I possibly know how many epochs were used in each model？



---Original---

**Comment by VirajBagal** [comment]
@youngzhou97qz  Thanks for the response. I am using your code completely, so I don't think there is any difference in mine and yours optimizers, learning rates. I am making no changes to models (used as it is from the code), so no difference in order of layers, dim sizes, weights sharing, dropout, etc. In the above comment you said 'ensemble methods', did you use ensemble of models for the 0.64 best score? How did you ensemble them? I thought the scores reported were from individual models.

**Comment by youngzhou97qz** [comment]
@liushengyan It depends. Generally, you can stop training when the loss function of the validation set no longer drops.

**Comment by youngzhou97qz** [comment]
@VirajBagal You can try to re-divide the training and validation set, and train with multiple models. When you make predictions on the test set, you can use hard voting to avoid the wrong choice of a single model.

**Comment by VirajBagal** [comment]
@youngzhou97qz  The 0.64 accuracy score that you have got, is it thru the above ensembling or single model score?

**Comment by youngzhou97qz** [comment]
@VirajBagal  It's through the above ensembling model.

**Comment by VirajBagal** [comment]
@youngzhou97qz  Thank you for clarifying. I really appreciate that. Can you please tell how many models you used per category for ensembling?

**Comment by youngzhou97qz** [comment]
@VirajBagal 3 models.

**Comment by VirajBagal** [comment]
@youngzhou97qz  Can you please tell the **individual**/**single** model best score that you could achieve for each category ?

**Comment by rezaeiii** [comment]
Can you give me  the Run.py file of the CGMVQA code?
I run the code and did not reach the accuracy of 0.64
please?
